### PR TITLE
feat: consume and interact with back-end user's API

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

Add interaction with Service API to create users.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This feature was tested locally and with unit testing.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

Jira task: https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-65

### CODE REVIEW

File: src/views/Form/index.tsx [hunk 1]
```diff
@@ -40,40 +113,108 @@ const Form = () => {
...
+          styling="bcgov-normal-blue btn buttonMargin"
+          disabled={loading}
+        />
+        <Button
+          onClick={resetForm}
+          label="Reset"
+          styling="bcgov-normal-white btn buttonMargin"
+          disabled={loading}
+        />
+        <Button onClick={goBack} label="Back to home" styling="bcgov-normal-white btn buttonMargin" />
```
Comment (mgaseta) [inline src/views/Form/index.tsx:214]: I think this button should follow the same pattern as the other in the page (each attribute on its own line).

**Review by mgaseta — CHANGES_REQUESTED** [review]
Good job! Congrats 🎉! It looks great, just some minor changes.

File: src/components/TextInput/index.tsx [hunk 2]
```diff
@@ -5,24 +5,44 @@ interface InputProps {
...
+        type="text"
+        id={id}
+        className={inputClasses}
+        onChange={onChangeHandler}
+        value={inputValue}
+        onKeyDown={onKeyDownHandler}
+        disabled={disabled}
+      />
+      <div data-testid="feedback-element"
+            className={isValid ? 'valid-feedback' : 'invalid-feedback'}>{ feedback }</div>
```
[the remark this answers is not in the document]
Comment (RMCampos) [reply src/components/TextInput/index.tsx:45]: No problem. Fixed! :)

File: src/components/UserTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      </thead>
+      <tbody>
+        {elements.map((item, idx) => (
+          // FIXME: dont't use 'idx' as key: https://reactjs.org/docs/lists-and-keys.html#keys
+          <tr key={idx.toString()}>
+            <th scope="row">{idx}</th>
+            <td>{item.firstName}</td>
+            <td>{item.lastName}</td>
+            <td>
+              <Button onClick={() => deleteFn(idx)} label="Delete-me" styling="bcgov-normal-white btn" />
```
[the remark this answers is not in the document]
Comment (RMCampos) [reply src/components/UserTable/index.tsx:29]: Hahah indeed.

File: src/views/Form/index.tsx (hunk 1 above)
Comment (RMCampos) [reply src/views/Form/index.tsx:214]: Thanks for noticing!

**Review by mgaseta — APPROVED** [review]
LGTM! Nice work 🎉!

**Pull request merged**